File: Java/Tier/src/Tier.java
public class Tier {
    public static void main(String[] args) {
        Hund h = new Hund("Hund", "Bello", "grau", "10", "Männlich", " 15kg", "wuff wuff");
        h.geräuschErzeugen(h.geräusch);
        h.schlafen(h.tierart);
        h.pfoteGeben();
        System.out.println(h.beine);

        Katze k = new Katze("Katze", "Fiona", "grau", "4", "Weiblich", "12.5", "Miau");
        k.geräuschErzeugen(k.geräusch);
        k.schlafen(k.tierart);
        k.schnurren();
    }

}

File: C++/Tier/tier.cpp
#include <iostream>
using namespace std;

class Tier
{

public:
    int age;
    string name;
    string race;
};

class Hund : Tier
{
public:
    int age;
    string name;
    string race;

    void bark()
    {
        cout << "wuff wuff";
    }
};

class Katze : Tier
{

public:
};

int main()
{
    Hund golden;
    golden.age = 1;
    golden.name = "bello";
    golden.race = "golden";

    golden.bark();

    return 0;
}